### ISSUE TITLE
Adjust left menu to work at 200% magnification

### ISSUE THREAD

**Issue by atsansone** [dart-lang/site-www#5791]
Hotlist: https://buganizer.corp.google.com/hotlists/5891206
Bug: https://buganizer.corp.google.com/issues/337893089

### Observed Results

When the zoom reached 200% can be noticed that the left menu cannot be closed once the user has opened it.

[Voiceover](http://screencast/cast/NjU2MDA0NzI2OTE1MDcyMHw0NjlmMzgwZC1hZA)

### Expected Results

The left menu should be able to be closed at high levels of zoom

### User Impact

The users that have sight issues and need a high level or zoom, can have a challenging situation since the whole page content cannot be seen correctly.

### Steps To Reproduce

1. Go to https://dart.dev/overview
1. Go to settings in google chrome and select the + in order to increase the zoom
1. Notice that the menu in the left cannot be closed when the page is at 200% zoom

### Relevant GAR Sheet

[A11Y Tracker](https://a11ytracker.corp.google.com/products/034ba691-ac78-4cc2-88c2-e06449cfd17e/edit/testresults?useCase=5uc12pk41e1hg&e=ProductLabels)

### GAR Details

GAR 1.16 (a)[Magnified](go/GAR-web#run-through-magnification) (b)[Zoom](go/GAR-web#run-through-page-zoom)

### Environment Details

* Machine: Macbook Pro/Chromebook
* Browser: Version 124.0.6367.93 (Official Build) (arm64)
* Screen reader: Voiceover/ChromeVox
* Bug filling time: 15 mins
